### PR TITLE
fix: prevent deleted instances from being recreated

### PR DESCRIPTION
**Pull request by ImPedro29**
## Summary

- create Redis instance keys atomically with `SET NX`
- update existing keys with `SET XX KEEPTTL` so late events cannot resurrect deleted instances
- delete keys directly and idempotently
- serialize runtime cleanup and Redis deletion with connection operations
- verify the instance still exists before generating a new client
- add regression tests for duplicate creation and delete/update races

## Validation

- `go test ./...`
- `go test -race ./repositories/instances ./server/controllers ./lib/whatsmiau`
- `go vet ./...`

## Rollout

This is the first PR in the deletion-consistency rollout.

### CODE REVIEW

File: lib/whatsmiau/whatsmeow.go [hunk 1]
```diff
@@ -568,14 +576,49 @@ func (s *Whatsmiau) Status(id string) (Status, error) {
...
 		zap.L().Warn("logout: client does not exist", zap.String("id", id))
 		return nil
 	}
 
 	s.deleteClient(id)
-	return s.deleteDeviceIfExists(ctx, client)
+	if err := s.deleteDeviceIfExists(ctx, client); err != nil {
+		return err
+	}
+	s.clearInstanceRuntimeState(id)
```
Comment (linmelx) [inline lib/whatsmiau/whatsmeow.go]: Mover `clearInstanceRuntimeState` para antes da verificação de erro, para que a limpeza dos caches não dependa do sucesso de `logoutUnlocked`

**Review by linmelx — CHANGES_REQUESTED** [review]

File: lib/whatsmiau/whatsmeow.go (hunk 1 above)
Comment (ImPedro29) [reply lib/whatsmiau/whatsmeow.go]: Feito no commit 8014f5c. Coloquei `clearInstanceRuntimeState` em um `defer` no início de `logoutUnlocked`, então a limpeza agora ocorre no sucesso, quando não há client e também quando `deleteDeviceIfExists` retorna erro. Removi ainda a chamada redundante em `Delete` e adicionei um teste de regressão para o early return sem client. Validação: `go test ./...` e `go test -race ./lib/whatsmiau ./server/controllers ./repositories/instances`.

**Review by linmelx — APPROVED** [review]
